### ISSUE TITLE
Elasticsearch Storage : javax.net.ssl.SSLHandshakeException

### ISSUE THREAD

**Issue by Balajinew** [openzipkin/zipkin#1833]
I am trying to connect elasticsearch which is hosted in k8s environment , But getting SSL handshake exception , Below I have mentioned my connection properties and error details , please provide if any resolution for this issue.

zipkin:
  storage:
    type: elasticsearch
elasticsearch:
      cluster: ${ES_CLUSTER:*********}
      hosts: ${ES_HOSTS:https://******:9200}
      index: ${ES_INDEX:zipkin}
      index-shards: ${ES_INDEX_SHARDS:5}
      index-replicas: ${ES_INDEX_REPLICAS:1}
      username: ${ES_USERNAME:****}
      password: ${ES_PASSWORD:******}

Error : 

ERROR: cannot load service names: Request processing failed; nested exception is java.io.UncheckedIOException: javax.net.ssl.SSLHandshakeException: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target

**Comment by codefromthecrypt** [comment]
Is your certificate self-signed?

**Comment by rangwea** [comment]
@adriancole I have this problem too.  How to solve this problem?

**Comment by codefromthecrypt** [comment]
@rangwea is your certificate self-signed?

**Comment by rangwea** [comment]
@adriancole yes，my certificate is self-signed. This is test environment.

**Comment by rangwea** [comment]
@adriancole Can I ignore the ssl validate by configuring.

**Comment by rangwea** [comment]
@adriancole Can I ignore the ssl validate by configuring.

**Comment by codefromthecrypt** [comment]
Sorry.. in java issues like this are usually remedied by modifying your keystore with the self-signed certificate or setting parameters like `javax.net.ssl.trustStore` and `javax.net.ssl.trustStorePassword` to override. Doing a little googling I found this, but other links should also work https://access.redhat.com/documentation/en-us/jboss_enterprise_application_platform/5/html/security_guide/ch15s02s02

**Comment by rangwea** [comment]
Thank. I have solved the problem by setting `javax.net.ssl.trustStore`.

At 2018-01-25 19:03:07, "Adrian Cole" <notifications@github.com> wrote:


Sorry.. in java issues like this are usually remedied by modifying your keystore with the self-signed certificate or setting parameters like javax.net.ssl.trustStore and javax.net.ssl.trustStorePassword to override. Doing a little googling I found this, but other links would also work https://access.redhat.com/documentation/en-us/jboss_enterprise_application_platform/5/html/security_guide/ch15s02s02
